zhaquirks/legrand/remote_switch.py

```diff
@@ -0,0 +1,212 @@
+"""Device handler for Legrand Remote switch."""
+# import logging
+from typing import Any, List, Optional, Union
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import (
+    Basic,
+    BinaryInput,
+    Identify,
+    LevelControl,
+    OnOff,
+    Ota,
+    PollControl,
+    PowerConfiguration,
+)
+from zigpy.zcl.clusters.manufacturer_specific import ManufacturerSpecificCluster
+
+from zhaquirks import Bus, PowerConfigurationCluster
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.legrand import LEGRAND
+
+MANUFACTURER_SPECIFIC_CLUSTER_ID = 0xFC01  # decimal = 64513
+
+VOLTAGE_REPORTED = "voltage_reported"
+
+
+class LegrandRemoteSwitchCluster(CustomCluster, ManufacturerSpecificCluster):
+    """LegrandRemoteSwitchCluster."""
+
+    cluster_id = MANUFACTURER_SPECIFIC_CLUSTER_ID
+    name = "LegrandRemoteSwitchCluster"
+    ep_attribute = "legrand_cluster"
+
+
+class LegrandRemoteSwitchLevelControl(CustomCluster, LevelControl):
+    """LegrandRemoteSwitchLevelControl."""
+
+    def bind(self):
+        """Disable binding to SwitchLevelControl cluster of the coordinator."""
+
+        pass
+
+
+class LegrandRemoteSwitchOnOff(CustomCluster, OnOff):
+    """Legrand Wireless Remote Switch On/Off custom cluster."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        self._current_state = {}
+        super().__init__(*args, **kwargs)
+
+    def handle_cluster_request(
+        self,
+        hdr: foundation.ZCLHeader,
+        args: List[Any],
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ):
+        """handle_cluster_request."""
+
+        if hdr.command_id in (0x0000, 0x0001, 0x0002):
+            # Send default response
+            if not hdr.frame_control.disable_default_response:
+                self.send_default_rsp(hdr, status=foundation.Status.SUCCESS)
+
+            self.endpoint.device.voltage_bus.listener_event(VOLTAGE_REPORTED)
+        else:
+            return super().handle_cluster_request(
+                hdr, args, dst_addressing=dst_addressing
+            )
+
+
+class LegrandRemoteSwitchPowerConfiguration(PowerConfigurationCluster):
+    """Legrand Wireless Remote Switch On/Off custom cluster."""
+
+    _CONSTANT_ATTRIBUTES = {
+        0x0031: 10,  # battery_size = CR2032
+        0x0033: 1,  # battery_quantity
+        0x0034: 30,  # battery_rated_voltage = 3000 mV
+    }
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Remove init

Reply:
https://github.com/zigpy/zha-device-handlers/blob/1274644d9a67a7334fca6fc96f831cfb23390761/zhaquirks/ecolink/contact.py#L21

As an example, aside from your constant attributes ( if they aren’t present on the device that is) this is all that’s needed and only if overriding the defaults it uses